### ISSUE TITLE
Support for IQuestLab/IQuest-Coder-V1-40B-Instruct

### ISSUE THREAD

**Issue by blackcat1402** [turboderp-org/exllamav3#134]
As is well known, DeepSeek's predecessor was a quantitative trading company in the Chinese stock market. Its biggest competitor in China, NineK, has also released a code model, very small at 40B param size, but claiming performance exceeding Sonnet 4.5. This architecture may be new, and hopefully Exllamav3 will provide support for this model. 
IQuestLab/IQuest-Coder-V1-40B-Instruct
https://huggingface.co/IQuestLab/IQuest-Coder-V1-40B-Instruct

**Comment by samueldemoura** [comment]
Leaving a +1 for the non-loop variant, and some unsolicited thoughts:

> The benchmarks have been adjusted after some https://github.com/IQuestLab/IQuest-Coder-V1/issues/14#issuecomment-3705756919 was discovered, so it no longer outperforms GPT5 and Sonnet, and I don't think a lot of people trust the adjusted score either.

FWIW, on the LocalLLaMA subreddit I saw some users thrashing the model as "utterly useless" after the reward hacking stuff came to light, but I played around with the non-loop GGUFs for a bit and the model actually seems decent.

It has undergone enough agentic RL and has good enough instruction following to be usable even with "non-native" tool calling (having a system prompt override the way the model formats its tool calls, basically). I couldn't test it on longer tasks due to ctxlen/VRAM limitations (llama.cpp's cache quantization generally destroys performance in agentic stuff in my experience, so I avoided it entirely for these tests), but going by vibes on the shorter tasks, it's no Sonnet 4.5, but I don't think the results are _100%_ benchmaxxing either.

> It would be interesting if anyone had benchmarks comparing the looping and non-looping variants of the model.

I haven't seen proper benchmarks, but [this Twitter user](https://x.com/waxhn/status/2006817815832166867) suggests that the loop stuff is "mostly cosmetic": you can take the loop model, ignore the loop logic entirely, and still get a 90%+ token match against a "proper" implementation during greedy decoding according to their tests.

Given:

- The decent performance of the non-loop version
- The fact that dense 40B is a pretty rare size for open models
- The fact that the team is releasing not just the base pretrain, but also midstage checkpoints
 
I think support for the regular version would definitely be valuable. Doubt the complexity of the loop version would be worth it though, especially given the latency/throughput hit on a model that's already kinda slow. If waxhn's investigation holds up, it might make more sense for someone to just try distilling logits from the loop version back into the normal model and make do with that instead 😁

**Comment by blackcat1402** [comment]
> Supported now in `dev'. First quant uploaded [here](https://huggingface.co/turboderp/IQuest-Coder-V1-40B-Instruct-exl3). The architecture is just Llama, and the only real obstacle was having to convert the tokenizer to HF format.
> 
> Catbench results are promising:
> 
> <img alt="Image" width="789" height="530" src="https://private-user-images.githubusercontent.com/11859846/531723292-640562d4-8e1e-4a73-a325-3bf39bf98aab.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njc1ODI1MDcsIm5iZiI6MTc2NzU4MjIwNywicGF0aCI6Ii8xMTg1OTg0Ni81MzE3MjMyOTItNjQwNTYyZDQtOGUxZS00YTczLWEzMjUtM2JmMzliZjk4YWFiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTA1VDAzMDMyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkzNTZlNzk5MTZjYmFjYTAyYTlhM2I0YmJlNjdlYzRlOGVjYWMyOGVkYWJjYmM5OGIxMDE5OWY3MmEyNzU2NmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.53XNaNeQbAAXsypF40gGrxH_trHde9qnwsvZcSgbWRk">

Thanks for your efforts. It looks not bad, LOL:D
